src/ImageSharp/Processing/Processors/Convolution/MedianBlurProcessor{TPixel}.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using System.Numerics;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Convolution
+{
+    /// <summary>
+    /// Applies an median filter.
+    /// </summary>
+    internal sealed class MedianBlurProcessor<TPixel> : ImageProcessor<TPixel>
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        private readonly MedianBlurProcessor definition;
+
+        public MedianBlurProcessor(Configuration configuration, MedianBlurProcessor definition, Image<TPixel> source, Rectangle sourceRectangle)
+            : base(configuration, source, sourceRectangle)
+        {
+            this.definition = definition;
+        }
+
+        protected override void OnFrameApply(ImageFrame<TPixel> source)
+        {
+            int kernelSize = (2 * this.definition.Radius) + 1;
+
+            MemoryAllocator allocator = this.Configuration.MemoryAllocator;
+            using Buffer2D<TPixel> targetPixels = allocator.Allocate2D<TPixel>(source.Width, source.Height);
+
+            source.CopyTo(targetPixels);
+
+            Rectangle interest = Rectangle.Intersect(this.SourceRectangle, source.Bounds());
+
+            // We use a rectangle with width set to 2 * kernelSize^2 + width, to allocate a buffer big enough
+            // for kernel source and target bulk pixel conversion.
+            Rectangle operationBounds = new Rectangle(interest.X, interest.Y, (2 * (kernelSize * kernelSize)) + interest.Width, interest.Height);
```

Comment:
If you use 2x interest width here you can use the extra buffer space to allow bulk per-row `ToVector4()` against the source row in the operation.

```c#
new Rectangle(interest.X, interest.Y, (2 * (kernelSize * kernelSize)) + (2 * interest.Width), interest.Height);
```

Reply:
Well, it's not that simple I think, as we need more rows for a single kernel.

I think I get your point about caching the `Vector4` converted pixels, I'll take the challenge of making it work!

Reply:
On the multiple source rows that Median needs, the conversion is now done with BulkOperations. I think this is what you meant.